lib/ignored-paths.js

```diff
@@ -81,64 +124,139 @@ function findIgnoreFile(cwd) {
  * IgnoredPaths
  * @constructor
  * @class IgnoredPaths
- * @param {Array} patterns to be matched against file paths
+ * @param {Object} options object containing 'ignore', 'ignorePath' and 'patterns' properties
  */
-function IgnoredPaths(patterns) {
-    this.patterns = patterns;
-}
+function IgnoredPaths(options) {
 
-/**
- * IgnoredPaths initializer
- * @param {Object} options object containing 'ignore' and 'ignorePath' properties
- * @returns {IgnoredPaths} object, with patterns loaded from the ignore file
- */
-IgnoredPaths.load = function(options) {
-    var patterns;
+    options = mergeDefaultOptions(options);
 
-    options = options || {};
+    /**
+     * Called when a possible os incompatibility was detected
+     * @param {string} message Hint to add to the compatibility warning
+     * @returns {void}
+     */
+    function printOsCompatibilityWarning(message) {
+        if (options.onOsCompatibilityIssue instanceof Function) {
+            options.onOsCompatibilityIssue([
+                "warning: two-globstar ignore pattern is not compatible with OSX.",
+                message
+            ].join(" "));
+        }
+    }
 
-    if (options.ignore) {
-        patterns = loadIgnoreFile(options.ignorePath || findIgnoreFile(options.cwd));
-    } else {
-        patterns = [];
+    /**
+     * add pattern to node-ignore instance
+     * @param {object} ig, instance of node-ignore
+     * @param {string} pattern, pattern do add to ig
+     * @returns {array} raw ignore rules
+     */
+    function addPattern(ig, pattern) {
+        if (pattern.indexOf("**") !== -1) {
+            printOsCompatibilityWarning(pattern);
+        }
+        return ig.addPattern(pattern);
     }
 
-    if (options.ignorePattern) {
-        patterns = patterns.concat(options.ignorePattern);
+    /**
+     * add ignore file to node-ignore instance
+     * @param {object} ig, instance of node-ignore
+     * @param {string} filepath, file to add to ig
+     * @returns {array} raw ignore rules
+     */
+    function addIgnoreFile(ig, filepath) {
+        var fileContent = fs.readFileSync(filepath, "utf8");
+        if (fileContent.indexOf("**") !== -1) {
+            printOsCompatibilityWarning("source: " + filepath);
```

Comment:
Fixing by replacing all occurrences `**` with a single star is obvious. Will remove the compatibility warnings.